src/components/Tree/Node.tsx

```diff
@@ -25,67 +27,89 @@ type NodeProps = {
+                            return;
+                        }
+
+                        if (onClick) {
+                            onClick(activeNodeId === id ? null : id);
+                        }
                     }}
                 >
-                    {nodes &&
-                        (showNodes ? (
-                            <IconCaretDown size={IconSize.Size16} />
-                        ) : (
-                            <IconCaretRight size={IconSize.Size16} />
-                        ))}
-                </span>
-                {icon && <span className="tw-pl-2">{icon}</span>}
-                <span className="tw-pl-2" data-test-id="node-link-name">
-                    {name}
-                </span>
-                <span
-                    className={merge([
-                        "tw-ml-auto tw-text-black-100 tw-text-opacity-40 tw-font-normal",
-                        selected && "tw-text-black-50",
-                    ])}
-                >
-                    {label}
-                </span>
-            </a>
+                    <div className="tw-flex tw-space-x-2 tw-items-center">
+                        <span
+                            data-test-id="toggle"
+                            onClick={(event) => {
+                                event.stopPropagation();
+                                setShowNodes(!showNodes);
+                            }}
+                        >
+                            {nodes &&
+                                (showNodes ? (
+                                    <IconCaretDown size={IconSize.Size16} />
+                                ) : (
+                                    <IconCaretRight size={IconSize.Size16} />
+                                ))}
+                        </span>
+                        {icon && <span>{icon}</span>}
+                        <span data-test-id="node-link-name">{name}</span>
+                    </div>
+                    <div className="tw-px-1.5">
+                        <span
+                            className={merge([
+                                "tw-text-black-100 tw-text-opacity-40 tw-font-normal",
+                                selected && "tw-text-black-50",
+                            ])}
+                        >
+                            {label}
+                        </span>
+                    </div>
+                </a>
+                <AnimatePresence>
+                    {actions && (isHovered || selected) && (
+                        <motion.div
+                            className="tw-flex tw-space-x-1.5 tw-items-center"
+                            initial={{ width: 0 }}
+                            animate={{ width: "auto" }}
+                            exit={{ width: 0 }}
+                            transition={{ ease: [0.04, 0.62, 0.23, 0.98] }}
+                        >
+                            {actions.map((action) => action)}
```

Comment:
why not forEach?

Reply:
TS complains that forEach has no return value, so map seems like a better solution than bending TS to our will :-)